G/git_crypt/build_tarballs.jl

```diff
@@ -0,0 +1,89 @@
+fi
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:i686, libc=:glibc),
+    Linux(:x86_64, libc=:glibc),
+    Linux(:aarch64, libc=:glibc),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:powerpc64le, libc=:glibc),
+    # # The following 4 commented out platforms need gcc7
+    # Linux(:i686, libc=:musl),
+    # Linux(:x86_64, libc=:musl),
+    Linux(:aarch64, libc=:musl),
+    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
+
+    # MacOS error:
+    # c++ -Wall -pedantic -Wno-long-long -O2 -std=c++11 -o git-crypt git-crypt.o commands.o crypto.o gpg.o key.o util.o parse_options.o coprocess.o fhstream.o crypto-openssl-10.o crypto-openssl-11.o -lcrypto
+    # Undefined symbols for architecture x86_64:
+    #   "_HMAC_CTX_free", referenced from:
+    #       __ZN15Hmac_sha1_stateD2Ev in crypto-openssl-11.o
+    #       __ZN15Hmac_sha1_stateD1Ev in crypto-openssl-11.o
+    #   "_HMAC_CTX_new", referenced from:
+    #       __ZN15Hmac_sha1_stateC2EPKhm in crypto-openssl-11.o
+    #       __ZN15Hmac_sha1_stateC1EPKhm in crypto-openssl-11.o
+    #   "_OPENSSL_init_crypto", referenced from:
+    #       __Z11init_cryptov in crypto-openssl-11.o
+    # ld: symbol(s) not found for architecture x86_64
+    # clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
+    # MacOS(:x86_64),
+
+    # Free BSD error:
+    # c++ -Wall -pedantic -Wno-long-long -O2 -std=c++11 -o git-crypt git-crypt.o commands.o crypto.o gpg.o key.o util.o parse_options.o coprocess.o fhstream.o crypto-openssl-10.o crypto-openssl-11.o -lcrypto
+    # crypto-openssl-11.o: In function `init_crypto()':
+    # crypto-openssl-11.cpp:(.text+0xd): undefined reference to `OPENSSL_init_crypto'
+    # crypto-openssl-11.o: In function `Hmac_sha1_state::Hmac_sha1_state(unsigned char const*, unsigned long)':
+    # crypto-openssl-11.cpp:(.text+0x262): undefined reference to `HMAC_CTX_new'
+    # crypto-openssl-11.o: In function `Hmac_sha1_state::~Hmac_sha1_state()':
+    # crypto-openssl-11.cpp:(.text+0x2d1): undefined reference to `HMAC_CTX_free'
+    # clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
+    # make: *** [Makefile:50: git-crypt] Error 1
+    # Previous command exited with 2
+    # FreeBSD(:x86_64),
+
+    Windows(:i686),
+    Windows(:x86_64),
+]
+gcc7_platforms = [
+    # The following platforms need gcc7
+    Linux(:i686, libc=:musl),
+    Linux(:x86_64, libc=:musl),
+]
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("git-crypt", :git_crypt)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    # PackageSpec(name="OpenSSL_jll", uuid="458c3c95-2e84-50aa-8efc-19380b2a3a95"), # BB can't handle this ATM
+    "OpenSSL_jll",
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
```

Comment:
When you have multiple `build_tarballs` do something like [what we did for Git](https://github.com/JuliaPackaging/Yggdrasil/blob/74355e0547077a289051e12edbd2e581adcfda33/G/Git/build_tarballs.jl#L76-L93).  But I wonder if we can just prefer GCC 7 everywhere (being C++ my concern here)

Reply:
Yea, that is why the i686-linux-musl and x86_64-linux-musl build fail on azure I think, but they work locally with gcc 4...

Reply:
What version of BB are you using locally?